### ISSUE TITLE
Tippy content is rendered before the popover is triggered?

### ISSUE THREAD

**Issue by reywright** [atomiks/tippyjs-react#288]
https://codesandbox.io/s/nervous-bassi-6ttfp

- **TLDR; I feel like children passed into Tippy should not render any of it's code before the popover is visible.** 
- There's a console.log here that fires that I don't expect to see before the popover is triggered.
-  If it makes more sense to fire this in the main Tippyjs repo I can do so.
- I'm happy to offer any elaboration or example code to explain further.

Many React components are built so that its lifecycle events / use effects / sideeffects fire when the component mounts. My assumption is that no part of a component should render if `visible` is false and that there would be some conditional in place based on whether it's visible or not.

I added a simple `Popover` component implementation to show this.... the console.log inside there does not fire unless it's visible.

**Comment by atomiks** [comment]
[Take a look here](https://gist.github.com/atomiks/520f4b0c7b537202a23a3059d4eec908). Seems to be a common enough issue for it to be default although I believe there was an integration issue with singletons..

**Comment by reywright** [comment]
Thanks @atomiks for following up! I created a codesandbox on that gist with the LazyTippy code you gave. There's an issue with `moveTransition` that is obvious in the codesandbox but I'm going to move forward with LazyTippy and axe usage of `moveTransition` for now. I'm happy to close this ticket or keep it or provide any further elaboration.

**Comment by atomiks** [comment]
#209
